src/osm_configurator/view/toplevelframes/calculation_frame.py

```diff
@@ -227,17 +279,105 @@ def __show_calculation_utilities(self):
+                                    border_width=button_constants_i.ButtonConstants.BUTTON_BORDER_WIDTH.value,
+                                    fg_color=button_constants_i.ButtonConstants.BUTTON_FG_COLOR_RED.value,
+                                    hover_color=button_constants_i.ButtonConstants.BUTTON_HOVER_COLOR.value,
+                                    border_color=button_constants_i.ButtonConstants.BUTTON_BORDER_COLOR.value,
+                                    text_color=button_constants_i.ButtonConstants.BUTTON_TEXT_COLOR.value,
+                                    command=self.__stop_calculation) \
+            .grid(column=4, row=1, rowspan=1, columnspace=1, padx=30, pady=40)
+
+        self.progressbar_label = \
+            customtkinter.CTkLabel(master=self.window,
+                                   text="Calculation started")\
+            .grid(column=2, row=1, rowspan=1, columnspace=1, padx=30, pady=10)
+
+        self.buttons.append(cancel_button)  # Add cancel button to buttons-list
+
+        self._calculation_controller.start_calculations(self._starting_point)  # starts the calculation from the chosen starting point
+
+        self.window.after(30000, self.__update_progressbar)  # keeps the progressbar up-to-date
+
+    def __update_progressbar(self) -> None:
+        """
+        Keeps the progressbar up-to-date. calls itself every five seconds if the calculation is not finished
+        """
+        calculation_state = self._calculation_controller.get_calculation_state()
+        calculation_phase = self._calculation_controller.get_current_calculation_phase()
+        calculation_process = self._calculation_controller.get_current_calculation_process()
+
+        if calculation_state == CalculationState.RUNNING and calculation_process < 1:
+            # Calculation is running and no phase change expected
+            self.progressbar.configure(value=calculation_process)
+            self.window.after(30000, self.__update_progressbar)
+            return
 
-        self.buttons.append(cancel_button)
+        if (calculation_state == CalculationState.RUNNING or calculation_state == CalculationState.ENDED_SUCCESSFULLY)\
+                and calculation_process == 1:
+            #  Phase change expected
+            if calculation_phase == CalculationPhase.AGGREGATION_PHASE:
+                # Calculation is done
+                self.__end_calculation()
+                return
+            else:
+                # State will be switched
+                calculation_phase = self.__get_next_phase(calculation_phase)
 
-        self._calculation_controller.start_calculations(self._starting_point)
+                self.progressbar.configure(value=0)  # Reset progressbar
+                self.progressbar_label.configure(text=calculation_phase.get_name())  # change label to the next phase
+
+                self.window.after(30000, self.__update_progressbar)
```

Comment:
recursion will lead to an error, since this recursion never stopps, or only stopps way to late